ccdproc/astrometry.py

```diff
@@ -0,0 +1,478 @@
+    -------
+    d : float or `numpy.ndarray`
+        Distance between x1,y1 and x2,y2
+    """
+    return ((x1-x2)**2 + (y1-y2)**2)**0.5
+
+
+def _get_index(z, n_stars):
+    """Return the indices from a set of permutations of length ``n_stars``.
+
+    For example, it will return the indices of the three stars
+    corresponding to the output from distance ratio.
+
+    Parameters
+    ----------
+    z : int
+        Index from 1-d permuation array
+
+    n_stars : int
+        Number of stars in the original array
+
+    Returns
+    -------
+    indices : tuple
+       i, j, k indices
+    """
+    count = 0
+    for i, j, k in it.permutations(range(0, n_stars), 3):
+        if count == z:
+            return i, j, k
+        count += 1
+
+
+def triangle_angle(a, b, c):
+    """Given the length of three sides, calculate
+       the angle of side a
+
+    Parameters
+    ----------
+    a : float
+       Side of a triangle
+
+    b : float
+       Side of a triangle
+
+    c : float
+       Side of a triangle
+
+
+    Returns
+    -------
+    theta : float
+       Angle of a triangle in radians
+    """
+    return np.arccos((b**2+c**2-a**2)/(2*b*c))
+
+
+def calc_triangle(x, y, i, j, k):
+    """From three points, create a description of a triangle
+
+    Parameters
+    ----------
+    x : `numpy.ndarray`
+        Array of x-positions
+
+    y : `numpy.ndarray`
+        Array of y-positions
+
+    i : int
+        index of first object
+
+    j : int
+        index of second object
+
+    k : int
+        index of third object
+
+    Returns
+    -------
+    sides : `numpy.ndarray`
+        Array of the length of each side of the triangle normalized to
+        the longest side
+
+    angles : `numpy.ndarray`
+        Array of the angles of each vertices of the triangle
+
+    order : `numpy.ndarray`
+        Order of the shortest to longest side
+
+    """
+    d1 = distance(x[i], y[i], x[j], y[j])
+    d2 = distance(x[i], y[i], x[k], y[k])
+    d3 = distance(x[j], y[j], x[k], y[k])
+    sides = np.array([d1, d2, d3])
+    a1 = triangle_angle(d1, d2, d3)
+    a2 = triangle_angle(d2, d1, d3)
+    a3 = triangle_angle(d3, d1, d2)
+    angles = np.array([a1, a2, a3])
+    order = sides.argsort()
+    return sides/sides.max(), angles, order
+
+
+def _calc_ratio(x, y, i, j, k):
+    """Calculate the ratio of the distance between the vertex at index i to
+    points j and k.
+
+    Parameters
+    ----------
+    x : `numpy.ndarray`
+        x-position of objects
+
+    y : `numpy.ndarray`
+        y-position of objects
+
+    i : int
+        Index of the vertex
+
+    j : int
+        Index for the first side
+
+    k : int
+        Index for the second side
+
+    Returns
+    -------
+    ratio: float
+        Ratio between the distance from point i to j to the distance between i
+        and k
+
+    """
+    d1 = distance(x[i], y[i], x[j], y[j])
+    d2 = distance(x[i], y[i], x[k], y[k])
+    if d2 == 0:
+        ratio = np.nan
```

Comment:
Given that you don't return `ratio` you could just remove this branch?

Reply:
Ah good catch, should probably just return np.nan which I think was suppose to happen.